dash.el

```diff
@@ -1591,6 +1591,54 @@ and when that result is non-nil, through the next form, etc."
                    (--> ,result ,form))
                  ,@more))))
 
+(defmacro -cond-> (x &rest clauses)
+  "Conditionally thread X through CLAUSES.
+Threads x (via `->') through each form for which the
+corresponding test expression is true.  Note that, unlike cond
+branching, `-cond->' threading does not short circuit after the
+first true test expression."
+  (declare (debug (form body))
+           (indent 1))
+  (when (-> clauses length (% 2) (= 1))
+    (error "Wrong number of arguments"))
+  (-let* ((g (dash--match-make-source-symbol))
+          (steps (-map
+                  (-lambda ((test step))
+                    `(if ,test
+                         (-> ,g ,step)
+                       ,g))
+                  (-partition 2 clauses))))
+    `(-let* ((,g ,x)
+             ,@(-zip-lists (-cycle (list g))
+                           (butlast steps)))
+       ,@(if (null steps)
+             g
+           (last steps)))))
+
+(defmacro -cond->> (x &rest clauses)
+  "Conditionally thread X through CLAUSES.
+Threads x (via `->>') through each form for which the
+corresponding test expression is true.  Note that, unlike cond
+branching, `-cond->>' threading does not short circuit after the
+first true test expression."
+  (declare (debug (form body))
+           (indent 1))
+  (when (-> clauses length (% 2) (= 1))
+    (error "Wrong number of arguments."))
```

Comment:
Same comments as with `-cond->`, but also error messages shouldn't end in a full stop; see [`(info "(elisp) Error Symbols")`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Error-Symbols.html).

The two macros also seem to be identical, bar the threading macro used. Can't we define them in terms of the same function for less duplication?

Reply:
Sorry about the full stop, never making this mistake again! 😄

The idea of defining a common structure has crossed my mind a few times, but since I set up to port these macros, i didn't pay much attention to it.